divi/src/coins.cpp

```diff
@@ -151,20 +151,53 @@ std::string CCoins::ToString() const
     return res.str();
 }
 
-CCoinsViewBacked::CCoinsViewBacked() : base(nullptr) {}
-CCoinsViewBacked::CCoinsViewBacked(CCoinsView* viewIn) : base(viewIn) {}
+CCoinsViewBacked::CCoinsViewBacked()
+  : base(nullptr), readonlyBase(false)
+{}
+
+CCoinsViewBacked::CCoinsViewBacked(CCoinsView* viewIn)
+  : base(viewIn), readonlyBase(false)
+{}
+
+CCoinsViewBacked::CCoinsViewBacked(const CCoinsView* viewIn)
+  : base(const_cast<CCoinsView*>(viewIn)), readonlyBase(true)
+{}
+
```

Comment:
Wouldn't it be safer to have two base pointer types, one const and one non-const and only the one used as an argument is initialized to be non-null? That way you can have a bit stronger type safety than with a boolean flag and assert on the pointer directly. It also guards against future extensions of the interfaces

Reply:
That's a neat idea, I've updated the code accordingly.